File: QtAdditions/src/QWidgetScrollListWidget.cpp
#include "dak/QtAdditions/QWidgetScrollListWidget.h"
#include "dak/QtAdditions/QWidgetListWidget.h"
#include "dak/QtAdditions/QWidgetListMimeData.h"

namespace dak::QtAdditions
{
   QWidgetScrollListWidget::QWidgetScrollListWidget(QWidget* widget, QWidget* parent)
   : QScrollArea(parent)
   {
      setWidget(widget);

      bool isVertical = true;
      if (auto list = dynamic_cast<QWidgetListWidget*>(widget))
         if (!list->isVertical())
            isVertical = false;

      if (isVertical)
      {
         setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
         setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
         setSizePolicy(QSizePolicy(QSizePolicy::Minimum, QSizePolicy::MinimumExpanding));
      }
      else
      {
         setHorizontalScrollBarPolicy(Qt::ScrollBarAsNeeded);
         setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
         setSizePolicy(QSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::Minimum));
      }

      setWidgetResizable(true);
      setSizeAdjustPolicy(SizeAdjustPolicy::AdjustToContents);
   }

}

File: PythonExample/qWidgetScrollListWidget.py
from PySide2.QtWidgets import QScrollArea,QWidget,QSizePolicy
from PySide2.QtCore import Qt


class QWidgetScrollListWidget(QScrollArea):
    def __init__(self,widget : QWidget, parent : QWidget = None):
        super().__init__(parent)
        self.setWidget(widget)
        self.isVertical = widget.isVertical()

        if self.isVertical:
            self.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
            self.setVerticalScrollBarPolicy(Qt.ScrollBarAsNeeded)
            self.setSizePolicy(QSizePolicy.Minimum,QSizePolicy.MinimumExpanding)
        else:
            self.setHorizontalScrollBarPolicy(Qt.ScrollBarAsNeeded)
            self.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
            self.setSizePolicy(QSizePolicy.MinimumExpanding,QSizePolicy.Minimum)

        self.setWidgetResizable(True)
        self.setSizeAdjustPolicy(self.sizeAdjustPolicy().AdjustToContents)
